File: src/NodeParser/StringTypeNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import type { BaseType } from "../Type/BaseType.js";
import { StringType } from "../Type/StringType.js";

export class StringTypeNodeParser implements SubNodeParser {
    public supportsNode(node: ts.KeywordTypeNode): boolean {
        return node.kind === ts.SyntaxKind.StringKeyword;
    }

    public createType(node: ts.KeywordTypeNode, context: Context): BaseType {
        return new StringType();
    }
}

File: internal/parser/string_type_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// StringTypeNodeParser handles the `string` keyword
// (src/NodeParser/StringTypeNodeParser.ts).
type StringTypeNodeParser struct{}

func NewStringTypeNodeParser() *StringTypeNodeParser {
	return &StringTypeNodeParser{}
}

func (p *StringTypeNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindStringKeyword
}

func (p *StringTypeNodeParser) CreateType(_ *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.StringType{}
}
